infra/graph_generators.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Sun Mar 17 13:00:33 2019
+
+@author: Zargham
+"""
+import networkx as nx
+
+def lineGraphGen(N, bidir=False,nodeTypeName='vanilla',edgeTypeName='vanilla'):
+
+    line = nx.path_graph(N, create_using=nx.MultiDiGraph)
+    if not(bidir):
+        G = line
+    else:  
+        edges = line.edges
+        G = nx.MultiDiGraph()
+        for e in edges:
+            G.add_edge(e[0],e[1])
+            G.add_edge(e[1],e[0])
+    
+    nx.set_node_attributes(G,nodeTypeName, 'type')
```

Comment:
if we do keep the nodeTypeName and edgeTypeName logic and arguments, we shouldn't keep repeating this logic all over the place. instead, we should have a `GraphOptions` type which includes a node type name and edge type name, and then a method like `_apply_graph_options(graph, options)` which does these mutations through a single re-used code path. then we can test it more thoroughly, easily change the logic, etc.